packages/core/admin/admin/src/content-manager/pages/ListView/components/TableRows/index.js

```diff
@@ -27,19 +27,64 @@ const TableRows = ({
   withBulkActions,
   rows,
 }) => {
-  const {
-    push,
-    location: { pathname },
-  } = useHistory();
+  const { push, location } = useHistory();
+  const { pathname } = location;
   const { formatMessage } = useIntl();
+  const { post } = useFetchClient();
 
   const { trackUsage } = useTracking();
   const pluginsQueryParams = usePluginsQueryParams();
 
+  /**
+   *
+   * @param {string} id
+   * @returns void
+   */
+  const handleRowClick = (id) => () => {
+    if (!withBulkActions) return;
+
+    trackUsage('willEditEntryFromList');
+    push({
+      pathname: `${pathname}/${id}`,
+      state: { from: pathname },
+      search: pluginsQueryParams,
+    });
+  };
+
+  const handleCloneClick = (id) => async () => {
+    try {
+      const { data } = await post(
+        `/content-manager/collection-types/${contentType.uid}/clone/${id}`
+      );
+
+      if ('id' in data) {
+        push({
+          pathname: `${pathname}/${data.id}`,
+          state: { from: pathname },
+          search: pluginsQueryParams,
+        });
+      }
+    } catch (err) {
+      if (err instanceof AxiosError) {
+        const serverError = err.response.data;
+
+        push({
+          pathname: `${pathname}/create/clone/${id}`,
+          state: { from: pathname, error: serverError },
+          search: pluginsQueryParams,
+        });
+      }
```

Comment:
What do you think about using `formatAPIError` already here and only pass the error message forward? This would simplify this part to

```suggestion
      push({
        pathname: `${pathname}/create/clone/${id}`,
        state: { from: pathname, error: formatAPIError(error) },
        search: pluginsQueryParams,
      });
```

and the other side needs to know less about what `error` looks like.